x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/snapshots/SourceOnlySnapshotIT.java

```diff
@@ -147,6 +151,41 @@ public void testSnapshotAndRestoreWithNested() throws Exception {
         assertHits(sourceIdx, builders.length, true);
     }
 
+    public void testSnapshotWithDanglingLocalSegment() throws IOException {
```

Comment:
This test could be made a little shorter and nicer with using #71411 so I only made it a best-effort reproducer here for now.